application/controllers/HostController.php

```diff
@@ -2,21 +2,54 @@
 
 namespace Icinga\Module\Businessprocess\Controllers;
 
-use Icinga\Module\Monitoring\Controller;
+use Icinga\Module\Businessprocess\Common\IcingadbDatabase;
+use Icinga\Module\Businessprocess\Web\Controller;
+use Icinga\Module\Icingadb\Model\Host;
+use Icinga\Module\Monitoring\Backend;
 use Icinga\Web\Url;
 
 class HostController extends Controller
 {
+    use IcingadbDatabase;
+
+    protected $backend;
+
+    protected $allParams;
+
+
     public function showAction()
     {
+        $this->allParams = $this->getAllParams();
+
         $host = $this->params->getRequired('host');
 
-        $query = $this->backend->select()
-            ->from('hoststatus', array('host_name'))
-            ->where('host_name', $host);
+        if (array_key_exists('backend', $this->allParams)) {
+            if ($this->allParams['backend'] === '_icingadb') {
+                $this->backend = $this->getDb();
```

Comment:
Don't use `$this->backend` here. That is the backend from the monitoring controller. Just use `$icingaDb = $this->getDb()` and work with that variable.